knockserver/views.py

```diff
@@ -6,8 +6,83 @@
 
 from knockserver import app
 from knockserver.database import db_session
-from knockserver.models import ExampleObject
+from knockserver.models import AccessPattern, PatternPiece
 
 @app.route('/')
 def index():
     return render_template('index.html')
+
+@app.route('/patterns/', methods=['GET', 'POST'])
+def patterns():
```

Comment:
To increase readability, I'd suggest breaking this into two functions. One for the `GET` (`def patterns_get()`) and one for the `POST` (`def patterns_post()`). Then it's one less `if` we need, and it moves that logic to the routing controller.